### PR TITLE
Study view - Survival plot with multiple curves ("rainbow")

### PR DESCRIPTION
**Pull request by adamabeshouse**

### CODE REVIEW

File: src/pages/resultsView/survival/SurvivalChart.tsx [hunk 1]
```diff
@@ -10,20 +10,29 @@ import {
...
+import {ClinicalAttribute} from "../../../shared/api/generated/CBioPortalAPI";
+import DefaultTooltip from "../../../shared/components/defaultTooltip/DefaultTooltip";
 
 export interface ISurvivalChartProps {
     alteredPatientSurvivals: PatientSurvival[];
     unalteredPatientSurvivals: PatientSurvival[];
+    patientToAnalysisGroup?:{[uniquePatientKey:string]:string};
+    analysisGroups?:ReadonlyArray<SurvivalAnalysisGroup>;
+    analysisClinicalAttribute?:ClinicalAttribute;
+    hideNACurve?:boolean; // if true, then hide analysis group data curve corresponding to value "NA"
```
Comment (zhx828) [inline src/pages/resultsView/survival/SurvivalChart.tsx]: This is probably no longer needed when we filter out the NA data

File: src/pages/resultsView/survival/SurvivalChart.tsx [hunk 2]
```diff
@@ -10,20 +10,29 @@ import {
...
 import {getPatientViewUrl} from "../../../shared/api/urls";
 import DownloadControls from "../../../shared/components/downloadControls/DownloadControls";
 import autobind from "autobind-decorator";
+import {SurvivalAnalysisGroup} from "../../studyView/StudyViewPageStore";
+import classnames from "classnames";
+import {ClinicalAttribute} from "../../../shared/api/generated/CBioPortalAPI";
+import DefaultTooltip from "../../../shared/components/defaultTooltip/DefaultTooltip";
 
 export interface ISurvivalChartProps {
     alteredPatientSurvivals: PatientSurvival[];
```
Comment (zhx828) [inline src/pages/resultsView/survival/SurvivalChart.tsx]: We should lump The altered/unaltered groups and group analysisGroups together

**Comment by adamabeshouse** [comment]
we have this tooltip issue
image

File: src/pages/resultsView/survival/SurvivalUtil.tsx [hunk 3]
```diff
@@ -265,4 +266,25 @@ function filterBasedOnCoordinates(filters: SurvivalPlotFilters, _val:ScatterData
         return true;
     }
     return false;
+}
+
+
+export const ALTERED_GROUP_VALUE = "Altered";
+export const UNALTERED_GROUP_VALUE = "Unaltered";
+
+export function getSurvivalChartDataByAltered(
```
Comment (alisman) [inline src/pages/resultsView/survival/SurvivalUtil.tsx]: by altered ____ ?

Comment (adamabeshouse) [reply src/pages/resultsView/survival/SurvivalUtil.tsx]: i guess `ByAlteredStatus` ?

**Comment by adamabeshouse** [comment]
https://github.com/cBioPortal/cbioportal-frontend/pull/1421/commits

**Pull request closed without merging**